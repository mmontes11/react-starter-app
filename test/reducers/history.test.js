import historyReducer from "reducers/history";

describe("reducers/history", () => {
  const initialState = [{ squares: Array(9).fill(null) }];

  it("has a default state", () => {
    expect(historyReducer(undefined, { type: "@init" })).toMatchSnapshot();
  });
  it("reduces nothing", () => {
    expect(historyReducer(initialState, { type: "WHATEVER" })).toMatchSnapshot();
  });
  it("reduces SET_HISTORY", () => {
    let state = initialState;
    const actions = [
      { type: "SET_HISTORY", squareIndex: 0, symbol: "X", step: 0 },
      { type: "SET_HISTORY", squareIndex: 3, symbol: "O", step: 1 },
      { type: "SET_HISTORY", squareIndex: 1, symbol: "X", step: 2 },
      { type: "SET_HISTORY", squareIndex: 4, symbol: "O", step: 3 },
      { type: "SET_HISTORY", squareIndex: 2, symbol: "X", step: 4 },
      { type: "SET_HISTORY", squareIndex: 5, symbol: "O", step: 5 },
    ];
    actions.forEach(action => {
      state = historyReducer(state, action);
      expect(state).toMatchSnapshot();
    });
  });
});
