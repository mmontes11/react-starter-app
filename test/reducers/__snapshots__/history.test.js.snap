// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/history has a default state 1`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
]
`;

exports[`reducers/history reduces SET_HISTORY 1`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
]
`;

exports[`reducers/history reduces SET_HISTORY 2`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
]
`;

exports[`reducers/history reduces SET_HISTORY 3`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
]
`;

exports[`reducers/history reduces SET_HISTORY 4`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      null,
      "O",
      "O",
      null,
      null,
      null,
      null,
    ],
  },
]
`;

exports[`reducers/history reduces SET_HISTORY 5`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      null,
      "O",
      "O",
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      "X",
      "O",
      "O",
      null,
      null,
      null,
      null,
    ],
  },
]
`;

exports[`reducers/history reduces SET_HISTORY 6`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      null,
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      null,
      "O",
      null,
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      null,
      "O",
      "O",
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      "X",
      "O",
      "O",
      null,
      null,
      null,
      null,
    ],
  },
  Object {
    "squares": Array [
      "X",
      "X",
      "X",
      "O",
      "O",
      "O",
      null,
      null,
      null,
    ],
  },
]
`;

exports[`reducers/history reduces nothing 1`] = `
Array [
  Object {
    "squares": Array [
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
      null,
    ],
  },
]
`;
